main.py

```
@@ -265,7 +317,11 @@ def extract(req: ExtractRequest):
     if not req.text.strip():
         raise HTTPException(status_code=400, detail="No text provided.")
     try:
-        knowledge_map = extract_knowledge_map(req.text, api_key=req.api_key)
+        src = source_intake.from_text(req.text)   # default min_text_length=1
+    except ParseEmpty as err:
+        raise HTTPException(status_code=400, detail=str(err))
```

Ruff (0.15.12)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)